ServerRoomMonitoring.Generator/Messaging/SensorQueue.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Text;
+using RabbitMQ.Client;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using ServerRoomMonitoring.Generator.Config;
+using ServerRoomMonitoring.Generator.Models;
+using ServerRoomMonitoring.Generator.Messaging;
+
+namespace ServerRoomMonitoring.Generator.Messaging
+{
+    public class SensorQueue : ISensorQueue
+    {
+        private readonly string _hostname;
+        private readonly string _password;
+        private readonly string _queueName;
+        private readonly string _username;
+        private IConnection _connection;
+
+        public SensorQueue(IOptions<RabbitConfig> rabbitMqOptions)
+        {
+            _queueName = rabbitMqOptions.Value.QueueName;
```

Comment:
Czyściej będzie chyba nie zapisywać tych 4 wartości z configu, tylko przekazać całą instancję configu do metody CreateConnection i po prostu tam odczytać to co potrzebne.